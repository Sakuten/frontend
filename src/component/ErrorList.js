import React from 'react'
import { observer } from 'mobx-react'

const ErrorList = ({list}) => (
  <div data-test='errorlist'>
    {
      list.map(c =>
        <div data-test='errorlist-error'>
          {JSON.stringify(c)}
        </div>
      )
    }
  </div>
)

export default observer(ErrorList)
