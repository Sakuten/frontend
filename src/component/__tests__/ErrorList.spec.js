import React from 'react'
import { shallow } from 'enzyme'
import ErrorList from '../ErrorList'

const setup = propOverrides => {
  const props = Object.assign({
    list: []
  }, propOverrides)

  const wrapper = shallow(<ErrorList {...props} />)

  return {
    props,
    wrapper,
  }
}

describe('components', () => {
  describe('ErrorList', () => {
    test('render', () => {
      const { wrapper } = setup()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
