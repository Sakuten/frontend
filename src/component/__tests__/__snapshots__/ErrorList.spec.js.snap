// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components ErrorList render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ErrorList
    list={
        Array [
            Object {
              "message": "Error 1",
            },
            Object {
              "message": "Error 2",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          data-test="errorlist-error"
>
          {"message":"Error 1"}
</div>,
        <div
          data-test="errorlist-error"
>
          {"message":"Error 2"}
</div>,
      ],
      "data-test": "errorlist",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": "{\\"message\\":\\"Error 1\\"}",
          "data-test": "errorlist-error",
        },
        "ref": null,
        "rendered": "{\\"message\\":\\"Error 1\\"}",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": "{\\"message\\":\\"Error 2\\"}",
          "data-test": "errorlist-error",
        },
        "ref": null,
        "rendered": "{\\"message\\":\\"Error 2\\"}",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            data-test="errorlist-error"
>
            {"message":"Error 1"}
</div>,
          <div
            data-test="errorlist-error"
>
            {"message":"Error 2"}
</div>,
        ],
        "data-test": "errorlist",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": "{\\"message\\":\\"Error 1\\"}",
            "data-test": "errorlist-error",
          },
          "ref": null,
          "rendered": "{\\"message\\":\\"Error 1\\"}",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": "{\\"message\\":\\"Error 2\\"}",
            "data-test": "errorlist-error",
          },
          "ref": null,
          "rendered": "{\\"message\\":\\"Error 2\\"}",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
